src/tfr_sensor/src/image_topic_wrapper.cpp

```diff
@@ -0,0 +1,77 @@
+/**
+ * wrapper for an image stream, allows the user to get the most recent image
+ * from that stream on demand through a service. 
+ *
+ * The names of the service and sensor stream are configurable by the user.
+ *
+ * Subscribed Topics:
+ * <camera_topic>: user suppplied
+ * Provided Services:
+ * <service_name>: user supplied
+ *
+ * Parameters:
+ * ~camera_topic: the camera topic to subscribe to (string, default: "")
+ * ~service_name: the name of the service to advertise (string, default: "")
+ * 
+ * Relevant Messages:
+ * tfr_msgs::WrappedImage (srv)
+ * */
+#include <ros/ros.h>
+#include <ros/console.h>
+#include <sensor_msgs/Image.h>
+#include <image_transport/image_transport.h>
+#include <tfr_msgs/WrappedImage.h>
+
+class ImageWrapper
+{
+    public:
+
+        ImageWrapper(ros::NodeHandle &n, const std::string &camera_topic,
+                const std::string &service_name)
+        {
+            image_transport::ImageTransport it{n};
+            subscriber = it.subscribe(camera_topic, 20, &ImageWrapper::set_current, this);
+            server = n.advertiseService(service_name, &ImageWrapper::get_current, this);
+        }
+        
+        ~ImageWrapper() = default;
+        ImageWrapper(const ImageWrapper&) = delete;
+        ImageWrapper& operator=(const ImageWrapper&) = delete;
+        ImageWrapper(ImageWrapper&&) = delete;
+        ImageWrapper& operator=(ImageWrapper&&) = delete;
+
+    private:
+
+        //subscription callback
+        void set_current(const sensor_msgs::ImageConstPtr &i)
```

Comment:
Current coding standard is `lowerCamelCase` for these methods

Reply:
yup